test/integration/ethereum/flashMintLeveraged.spec.ts

```diff
@@ -0,0 +1,466 @@
+                  expect(inputSpent.lte(subjectMaxAmountIn)).to.be.true;
+                });
+
+                it("should quote the correct input amount", async () => {
+                  const inputBalanceBefore =
+                    inputTokenName == "ETH"
+                      ? await owner.wallet.getBalance()
+                      : await inputToken.balanceOf(owner.address);
+                  await subject();
+                  const inputBalanceAfter =
+                    inputTokenName == "ETH"
+                      ? await owner.wallet.getBalance()
+                      : await inputToken.balanceOf(owner.address);
+                  const inputSpent = inputBalanceBefore.sub(inputBalanceAfter);
+
+                  const quotedInputAmount = await subjectQuote();
+
+                  expect(quotedInputAmount).to.gt(preciseMul(inputSpent, ether(0.99)));
+                  expect(quotedInputAmount).to.lt(preciseMul(inputSpent, ether(1.01)));
+                });
+              },
+            );
+
+            describe(
+              inputTokenName == "ETH" ? "redeemExactSetForETH" : "#redeemExactSetForERC20",
+              () => {
+                let swapDataCollateralToDebt: SwapData;
+                let swapDataOutputToken: SwapData;
+
+                let outputToken: StandardTokenMock | IWETH;
+
+                let subjectSetToken: Address;
+                let subjectMinAmountOut: BigNumber;
+                let subjectOutputToken: Address;
+
+                async function subject() {
+                  if (inputTokenName == "ETH") {
+                    return flashMintLeveraged.redeemExactSetForETH(
+                      subjectSetToken,
+                      subjectSetAmount,
+                      subjectMinAmountOut,
+                      swapDataCollateralToDebt,
+                      swapDataOutputToken
+                    );
+                  }
+                  return flashMintLeveraged.redeemExactSetForERC20(
+                    subjectSetToken,
+                    subjectSetAmount,
+                    subjectOutputToken,
+                    subjectMinAmountOut,
+                    swapDataCollateralToDebt,
+                    swapDataOutputToken
+                );
+                }
+
+                async function subjectQuote(): Promise<BigNumber> {
+                  return flashMintLeveraged.callStatic.getRedeemExactSet(
+                    subjectSetToken,
+                    subjectSetAmount,
+                    swapDataCollateralToDebt, swapDataOutputToken
+                  );
+                }
+
+                beforeEach(async () => {
+                  swapDataCollateralToDebt = {
+                    path: [collateralTokenAddress, addresses.tokens.weth],
+                    fees: [500],
+                    pool: ADDRESS_ZERO,
+                    exchange: Exchange.UniV3,
+                  };
+
+                  if (inputTokenName == "collateralToken") {
```

Comment:
```suggestion
                  if (inputTokenName === "collateralToken") {
```

Reply:
resolved in:

https://github.com/IndexCoop/index-coop-smart-contracts/pull/143/commits/15b7fac84f9f9ffcdbc1ae0ca3c90bf27af95a36